spec/index.bs

```diff
@@ -7575,31 +7575,38 @@ GPUQueue includes GPUObjectBase;
 
             **Returns:** {{undefined}}
 
-            - If |source|.{{GPUImageCopyExternalImage/source}}
-                <l spec=html>[=is not origin-clean=]</l>,
+            1. Let |sourceImage| be |source|.{{GPUImageCopyExternalImage/source}}
+            1. If |sourceImage| <l spec=html>[=is not origin-clean=]</l>,
                 throw a {{SecurityError}} and stop.
-
-            If any of the following requirements are unmet, throw an {{OperationError}} and stop.
-            <div class=validusage>
-                - Let |textureDesc| be |destination|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/[[descriptor]]}}.
-                - |copySize|.[=Extent3D/depthOrArrayLayers=] must be `1`.
-                - |textureDesc|.{{GPUTextureDescriptor/usage}} must include both {{GPUTextureUsage/RENDER_ATTACHMENT}} and {{GPUTextureUsage/COPY_DST}}.
-                - |textureDesc|.{{GPUTextureDescriptor/dimension}} must be {{GPUTextureDimension/"2d"}}.
-                - |textureDesc|.{{GPUTextureDescriptor/format}} must be one of the following:
-                    - {{GPUTextureFormat/"rgba8unorm"}}
-                    - {{GPUTextureFormat/"rgba8unorm-srgb"}}
-                    - {{GPUTextureFormat/"bgra8unorm"}}
-                    - {{GPUTextureFormat/"bgra8unorm-srgb"}}
-                    - {{GPUTextureFormat/"rgb10a2unorm"}}
-                    - {{GPUTextureFormat/"rgba16float"}}
-                    - {{GPUTextureFormat/"rgba32float"}}
-                    - {{GPUTextureFormat/"rg8unorm"}}
-                    - {{GPUTextureFormat/"rg16float"}}
-            </div>
-
-            Issue: The above list was written just for ImageBitmap.
-            Re-evaluate it for canvas (and video).
-            (It's probably still fine, though `rgb10a2unorm` is a bit of an outlier.)
+            1. If |sourceImage| is an {{ImageBitmap}} and
```

Comment:
Is there the same validation requirements for OffscreenCanvas?

Reply:
Good catch, OffscreenCanvas should be detachable as well but it doesn't seem to be detachable in the same way as a "platform object". I found this which I guess we can simply call into:
https://html.spec.whatwg.org/multipage/canvas.html#check-the-usability-of-the-image-argument
(and if that's missing checks for detachedness, that should be fixed upstream)

Notably, using that algorithm means that using zero-sized HTMLCanvasElement/OffscreenCanvas becomes an error (even if the copy size is zero). I guess we should adopt that since it's consistent with other usages.

> If `image` has either a horizontal dimension or a vertical dimension equal to zero, then throw an "InvalidStateError" DOMException.